internal/server/manager.go

```diff
@@ -37,18 +38,32 @@ type ClusterInfo struct {
 	FailureMessage *string           `json:"failureMessage"`
 }
 
-type instance struct {
-	client     client.Client
-	scheme     *runtime.Scheme
-	clusterMux sync.Mutex // use a Mutex to update managed Clusters
+type ClusterProfileStatus struct {
+	Name        *string                         `json:"name"`
+	Namespace   *string                         `json:"namespace"`
+	ClusterName *string                         `json:"clusterName"`
+	Summary     []configv1alpha1.FeatureSummary `json:"summary"`
+}
 
-	sveltosClusters map[corev1.ObjectReference]ClusterInfo
-	capiClusters    map[corev1.ObjectReference]ClusterInfo
+type instance struct {
+	client             client.Client
+	scheme             *runtime.Scheme
+	clusterMux         sync.Mutex // use a Mutex to update managed Clusters
```

Comment:
we need, a s you pointed out, those locks to be sync.RWMutex so API calls (reads) can all go in parallel.

Reply:
I totally forgot that, thank you